### ISSUE TITLE
InputBarAccessoryView slides to top on device with SwiftUI blocking messages

### ISSUE THREAD

**Issue by e1block** [MessageKit/MessageKit#1734]
**Describe the bug**

When you come to enter a message, the `InputBarAccessoryView` pushes up so that the keyboard can be shown. This works correctly in the Simulator, but not on device. The messages get blocked in behind. See screenshot attached.

**To Reproduce**

Run the Example app on device, select **SwiftUI Example**, try to send a message.

**Expected behavior**

The messages are shown _above_ the `InputBarAccessoryView` like they do with the other examples or when running in the Simulator.

**Screenshots**

![Screen Shot 2022-08-10 at 12 46 37 PM](https://user-images.githubusercontent.com/104395039/183981210-56f7a400-5f96-4d33-9f61-0981f18f8f86.png)

**Environment**
- What version of MessageKit are you using? **Latest main branch** - https://github.com/MessageKit/MessageKit/commit/14dd6aa335e76c46a06db7e9ee5c1d62a0ccca6a
- What version of iOS are you running on? **15.4.1**
- What version of Swift are you running on? **Swift 5**
- What device(s) are you testing on? Are these simulators? **iPhone 11 Pro (device), iPhone 13 Pro (simulator)**
- Is the issue you're experiencing reproducible in the example app? **Yes**

**Comment by e1block** [comment]
It actually occurs in the simulator too when you have the software keyboard actually showing (setting). From what I can tell, the issue is in `MessagesViewController.updateMessageCollectionViewBottomInset()`. In the case of Swift, the `collectionViewHeight` needs to be the parent view since there's this wrapper structure between SwiftUI & UIKit. Still debugging.

**Comment by martinpucik** [comment]
Thanks for raising this issue @e1block and trying to debug this. SwiftUI support is experimental and the main focus is for UIKit based applications.

**Comment by e1block** [comment]
For sure. Nonetheless, any insights would be appreciated.

**Comment by e1block** [comment]
Looks like the root of the bug is in [InputBarAccessoryView](https://github.com/nathantannar4/InputBarAccessoryView). The SwiftUI example there also has an incorrect offset as compared to the UIKit examples.

**Comment by Kaspik** [comment]
Closing in favor of the same issue in IBAV.

**Comment by e1block** [comment]
Figured it out.

`.ignoresSafeArea(.keyboard, edges: .bottom)` on the SwiftUI side. On the `UIViewControllerRepresentable` instance of `MessagesViewController`. This disables the builtin keyboard avoidance that SwiftUI has, since MessageKit has its own already.

Will make a PR on the demo code when I get a chance.

**Comment by bart-kneepkens** [comment]
> Figured it out.
> 
> `.ignoresSafeArea(.keyboard, edges: .bottom)` on the SwiftUI side. On the `UIViewControllerRepresentable` instance of `MessagesViewController`. This disables the builtin keyboard avoidance that SwiftUI has, since MessageKit has its own already.
> 
> Will make a PR on the demo code when I get a chance.

Thanks for sharing, but for me (iPhone X, iOS 16) this will still leave a gap below the `messageInputBar` (albeit smaller than before).

**Comment by e1block** [comment]
@bart-kneepkens Indeed, I'm seeing the same. Were you able to figure it out?

**Comment by bart-kneepkens** [comment]
@e1block unfortunately not really, unrelated to this I decided to move to UIKit navigation and that also resolves the issue 🤷‍♂️

**Comment by e1block** [comment]
Nice. Do you have some reference code to share on how you did that? Are you no longer using `UIViewControllerRepresentable` ?

**Comment by bart-kneepkens** [comment]
@e1block  I'm not at liberty of sharing the code, but in general it's very simple. I took inspiration from [this blog post](https://augmentedcode.io/2022/03/07/uikit-navigation-with-swiftui-views/). The app has a central tab bar controller that's manipulated by a coordinator. UIViewControllerRepresentable is not used much, but UIHostingController is.

**Comment by e1block** [comment]
Great, thanks.

**Comment by scottishwilly90009** [comment]
Send me the code and purpose

**Comment by Janneman84** [comment]
The conclusion of the mention is that you need do two things to fix the issue:
- use the latest InputBarAccessoryView, the necessary fix isn't part of a release yet so you'll need to use the master branch
- use `.ignoresSafeArea(.keyboard, edges: .bottom)`

**Comment by Janneman84** [comment]
The example app should have `SwiftUIExampleView.swift` be updated with the modifier.:

```swift
  var body: some View {
    MessagesView(messages: $messages).onAppear {
      self.connectToMessageSocket()
    }.onDisappear {
      self.cleanupSocket()
    }
    .navigationBarTitle("SwiftUI Example", displayMode: .inline)
    .modifier(IgnoresSafeArea()) //ensures correct input bar positioning
  }
    
  struct IgnoresSafeArea: ViewModifier {
      func body(content: Content) -> some View {
          if #available(iOS 14.0, *) {
              content.ignoresSafeArea(.keyboard, edges: .bottom)
          } else {
              content
          }
      }
  }
```

Also the documentation/faq should be updated to mention this too. Currently the faq just says SwiftUI is only experimental and isn't really supported.

So if anyone feels like making a PR be my guest :).
